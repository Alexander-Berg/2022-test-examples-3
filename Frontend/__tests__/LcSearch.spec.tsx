import * as React from 'react';
import { shallow } from 'enzyme';
import * as serializer from 'jest-serializer-html';
import { LcSearch } from '../LcSearch';
import { LcSizes, LcAlign } from '../../lcTypes/lcTypes';

expect.addSnapshotSerializer(serializer);

describe('<LcSearch/> component', () => {
    test('should render LcSearch', () => {
        const component = shallow(
            <LcSearch
                align={LcAlign.CENTER}
                size={LcSizes.M}
                width={LcSizes.S}
                placeholder="Поиск в Маркете"
                websiteUrl="market.yandex.ru"
                searchUrl="https://yandex.com.tr/search/"
                hasYellowBorder
                sectionId="1"
                events={[]}
                offsets={{
                    padding: {
                        top: LcSizes.S,
                        bottom: LcSizes.S,
                    },
                }}
            />
        );

        expect(component.html()).toMatchSnapshot();
    });
});
