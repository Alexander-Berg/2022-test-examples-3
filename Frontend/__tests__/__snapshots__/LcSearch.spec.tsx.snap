// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LcSearch/> component should render LcSearch 1`] = `

<section id
         class="lc-section lpc-section "
>
  <div class="lc-offsets lc-offsets_padding-top_s lc-offsets_padding-bottom_s">
    <div class="lc-search lc-search_align_center lc-search_yellow Theme Theme_color_yandex-default Theme_root_default">
      <div class="lc-search__content lc-search__content_width_s">
        <form role="search"
              method="GET"
              action="https://yandex.com.tr/search/"
        >
          <input type="hidden"
                 name="site"
                 value="market.yandex.ru"
          >
          <span class="Textinput Textinput_iconRight Textinput_size_m Textinput_view_default">
            <span class="Icon Textinput-Icon Textinput-Icon_side_right lc-search__button-wrapper"
                  aria-hidden="true"
            >
              <button class="lc-search__search-button"
                      type="submit"
              >
                <div class="lc-icon lc-icon_fill-type_fill lc-search__search-icon"
                     size="16"
                >
                  <svg width="16"
                       height="16"
                  >
                    <use xlink:href="#lc-icon__search">
                    </use>
                  </svg>
                </div>
              </button>
            </span>
            <input type="search"
                   aria-invalid="false"
                   autocomplete="off"
                   class="Textinput-Control"
                   name="text"
                   placeholder="Поиск в Маркете"
                   value
            >
            <span class="Textinput-Box">
            </span>
          </span>
        </form>
      </div>
    </div>
  </div>
</section>

`;
